libs/slides/src/lib/deck/deck.component.ts

```diff
@@ -51,18 +54,45 @@ export class SlidesDeckComponent {
   }
 
   nextSlide() {
-    this.goToSlide(this.activeSlideIndex + 1);
+    if (this.activeSlideIndex + 1 < this.slides.length) {
+      this.goToSlide(this.activeSlideIndex + 1);
+    } else if (this.nextLink != null && this.nextLink !== '') {
+      this.router.navigateByUrl(this.nextLink);
+    }
   }
 
   previousSlide() {
-    this.goToSlide(this.activeSlideIndex - 1);
+    if (this.activeSlideIndex > 0) {
+      this.goToSlide(this.activeSlideIndex - 1);
+    } else if (this.previousLink != null && this.previousLink !== '') {
+      this.router.navigateByUrl(this.previousLink);
+    }
   }
 
   canGoNext(): boolean {
-    return this.activeSlideIndex + 1 < this.slides.length;
+    return this.activeSlideIndex + 1 < this.slides.length || (this.nextLink != null && this.nextLink !== '');
   }
 
   canGoPrevious(): boolean {
-    return this.activeSlideIndex > 0;
+    return this.activeSlideIndex > 0 || (this.previousLink != null && this.previousLink !== '');
+  }
+
+  public setupPreviousNext(allRoutes: string[]) {
```

Comment:
Let's keep the presentation simple:
setNextLink()
and
setNextLink()
All the router manipulations should be done outside of this component, as it is not supposed to know about the codelab routing structure